packages/react-dom/src/__tests__/ReactDOMInput-test.js

```diff
@@ -167,6 +167,57 @@ describe('ReactDOMInput', () => {
     document.body.removeChild(container);
   });
 
+  it('should handle when initial state is 0', () => {
+    class App extends React.Component {
+      inputRef = null;
+      constructor(props) {
+        super(props);
+        this.state = {
+          value: 0,
+        };
+      }
+
+      render() {
+        return (
+          <div>
+            <h1>Hello, {this.props.name}</h1>
+            <input
+              type="number"
+              value={this.state.value}
+              ref={n => (this.inputRef = n)}
+              onChange={e =>
+                this.setState({
+                  value: parseFloat(e.target.value)
+                    ? parseFloat(e.target.value)
+                    : 0,
+                })
+              }
+            />
+          </div>
+        );
```

Comment:
Could you remove the wrapping `<div>` and `<h1>` tags? That way it's clearer what this component is trying to test.

Reply:
Yup sure, make sense.